pyro/distributions/distribution.py

```diff
@@ -111,4 +111,24 @@ def enumerate_support(self, expand=True):
         :return: An iterator over the distribution's discrete support.
         :rtype: iterator
         """
-        raise NotImplementedError("Support not implemented for {}".format(type(self)))
+        raise NotImplementedError("Support not implemented for {}".format(type(self).__name__))
+
+    def conjugate_update(self, other):
+        """
+        Creates an updated distribution fusing information from another
+        compatible distribution. This is supported by only a few conjugate
+        distributions.
+
+        This should satisfy::
+
+            fg, log_normalizer = f.conjugate_update(g)
+            assert f.log_prob(x) + g.log_prob(x) == fg.log_prob(x) + log_normalizer
+
+        :param other: A distribution representing ``p(data|self.probs)`` but
+            normalized over ``self.probs`` rather than ``data``.
```

Comment:
It took me a while to understand what's going on here... (it is not easy to interpret your `likelihood` terms in the tests). Nice idea! 

nit: probably you want to replace `self.probs` by `latent`.

Reply:
Thanks for reviewing! For intuition those likelihood terms in the test are "conjugated" i.e. "transposed-and-normalized" likelihoods, which are easier to interpret as Funsors (see Triaged comment in this PR description).

Reply:
Done, I've also added a dosctring note on the relation to `Funsor.__add__()`.

Reply:
Understood!